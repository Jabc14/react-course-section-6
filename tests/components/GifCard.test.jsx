import { render, screen } from "@testing-library/react";
import { GifCard } from "../../src/components/GifCard";

describe("Prueba a GifCard", () => {
  const title = "gta";
  const url = "https://www.google.com/";
  test("debe match con el snapshot", () => {
    const { container } = render(<GifCard title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Mostrar la img con el url y alt", () => {
    render(<GifCard title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Mostrar el titulo", () => {
    render(<GifCard title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
