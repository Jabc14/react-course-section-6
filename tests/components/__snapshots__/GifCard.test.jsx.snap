// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba a GifCard debe match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="gta"
      src="https://www.google.com/"
    />
    <p>
      gta
    </p>
  </div>
</div>
`;
